aspnet/mvc/models/formatting.rst

```diff
@@ -1,8 +1,223 @@
+
+Returning JSON-formatted data:
+
+.. literalinclude:: formatting/sample/src/ResponseFormattingSample/Controllers/Api/AuthorsController.cs
+  :language: c#
+  :lines: 21-26
+  :emphasize-lines: 3,5
+  :dedent: 8
+
+Sample response from this action:
+
+.. image:: formatting/_static/json-response.png
+
+Note that the content type of the response is ``application/json``, shown both in the list of network requests and in the Response Headers section. Also note the list of options presented by the browser (in this case, Microsoft Edge) in the Accept header, in the Request Headers section. The current technique is ignoring this header; obeying it is discussed below.
+
+To return plain text formatted data, use ``ContentResult`` and the ``Content`` helper:
+
+.. literalinclude:: formatting/sample/src/ResponseFormattingSample/Controllers/Api/AuthorsController.cs
+  :language: c#
+  :lines: 47-52
+  :emphasize-lines: 3,5
+  :dedent: 8
+
+A response from this action:
+
+.. image:: formatting/_static/text-response.png
+
+Note in this case the ``Content-Type`` returned is ``text/plain``. You can also achieve this same behavior using just a string response type:
+
+.. literalinclude:: formatting/sample/src/ResponseFormattingSample/Controllers/Api/AuthorsController.cs
+  :language: c#
+  :lines: 54-59
+  :emphasize-lines: 3,5
+  :dedent: 8
+
+.. tip:: For non-trivial actions with multiple return types or options (for example, different HTTP status codes based on the result of operations performed), prefer ``IActionResult`` as the return type.
+
+Content Negotiation
+-------------------
+
+Content negotiation (*conneg* for short) occurs when the client specifies an `Accept header <https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html>`_. The default format used by ASP.NET Core MVC is JSON. Negotiation occurs automatically for ``IActionResult`` return types, including results returned using the helper methods ``Ok``, ``BadRequest``, ``Created``, etc. You can also return a model type (a class you've defined as your data transfer type) and the framework will automatically wrap it in an ``ObjectResult`` for you.
+
+The following action method uses the ``Ok`` and ``NotFound`` helper methods:
+
+.. literalinclude:: formatting/sample/src/ResponseFormattingSample/Controllers/Api/AuthorsController.cs
+  :language: c#
+  :lines: 28-38
+  :emphasize-lines: 8,10
+  :dedent: 8
+
+A JSON-formatted response will be returned unless another format was requested and the server can return the requested format. You can use a tool like `Fiddler <http://www.telerik.com/fiddler>`_ to create a request that includes an Accept header, and specify another format. In that case, if the server has a *formatter* that matches the requested format, the results will be returned in that format.
```

Comment:
>if the server has a *formatter* that matches the requested format
If the server has a *formatter* that can produce a response in the requested format, the result will be returned in the format preferred by the client. (or client preferred format).